### PR TITLE
PR: Improve UI/UX of how repositioning panes work

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- Add new action to the Options menu of all dockable plugins called `Unlock position`, so that users can more easily find how to reorganize plugins in the interface.

    image

- Improve UI of title bar that is shown when `Unlock position` is clicked:
    - Make it taller.
    - Change icon for drag and drop area to better convey the idea that panes can be moved up/down or left/right to be repositioned.
    - Use close hand cursor when dragging pane around.
    - Add lock icon to the right to easily hide the title bar. Also use regular cursor when hovering on it.
    - Use hover and clicked palette states when interacting with it.
    - Add tooltip to the drag/drop and close areas.
    
    drag

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 ! Checking locally on Windows and the CI tests I noticed a couple of things:

* The new `Unlock position` is disabled for Plots:

plots


* Seems like the Options menu now positions different if there is no space to show all the actions. For example with the IPython Console options menu it overlaps the console instead of being shown at the top of the pane:

menu

* Some tests are failing due to the new action (`spyder/plugins/history/tests/test_plugin.py::test_init`) or due to missing setup code in the test at the end of the files like for the `EditorPluginExample`: https://github.com/spyder-ide/spyder/blob/504e7c77a9caf076d4b33961a7da9dee3fd3808b/spyder/plugins/editor/widgets/editor.py#L3559-L3585

File: spyder/api/widgets/main_widget.py [hunk 1]
```diff
@@ -338,6 +339,13 @@ def _setup(self):
             icon=self.create_icon('dock'),
             triggered=self.close_window,
         )
+        self.lock_unlock_action = self.create_action(
+            name=PluginMainWidgetActions.LockUnlockPosition,
+            text=_("Unlock position"),
+            tip=_("Unlock to move to another position"),
```
Comment (isabela-pf) [inline spyder/api/widgets/main_widget.py:345]:
```suggestion
            tip=_("Unlock to move pane to another position"),
```

File: spyder/api/widgets/main_widget.py [hunk 2]
```diff
@@ -418,6 +427,29 @@ def _on_top_level_change(self, top_level):
...
+            self.lock_unlock_action.setText(_('Lock position'))
+            self.lock_unlock_action.setIcon(self.create_icon('lock_open'))
+            self.lock_unlock_action.setToolTip(
+                _("Unlock to move to another position"))
+        else:
+            self.lock_unlock_action.setText(_('Unlock position'))
+            self.lock_unlock_action.setIcon(
+                self.create_icon('drag_dock_widget'))
+            self.lock_unlock_action.setToolTip(
+                _("Lock to the current position"))
```
Comment (isabela-pf) [inline spyder/api/widgets/main_widget.py:451]:
```suggestion
                _("Lock pane to the current position"))
```

File: spyder/plugins/base.py [hunk 3]
```diff
@@ -199,6 +199,14 @@ def __init__(self, parent=None):
             tip=_("Dock the pane"),
             triggered=self._close_window)
 
+        self._lock_unlock_action = create_action(
+            self,
+            text=_("Unlock position"),
+            tip=_("Unlock to move to another position"),
```
Comment (isabela-pf) [inline spyder/plugins/base.py:205]:
```suggestion
            tip=_("Unlock to move pane to another position"),
```

File: spyder/plugins/base.py [hunk 4]
```diff
@@ -479,3 +491,24 @@ def _add_dockwidget(self):
...
+        if visible:
+            self._lock_unlock_action.setText(_('Lock position'))
+            self._lock_unlock_action.setIcon(ima.icon('lock_open'))
+            self._lock_unlock_action.setToolTip(
+                _("Unlock to move to another position"))
+        else:
+            self._lock_unlock_action.setText(_('Unlock position'))
+            self._lock_unlock_action.setIcon(ima.icon('drag_dock_widget'))
+            self._lock_unlock_action.setToolTip(
+                _("Lock to the current position"))
```
Comment (isabela-pf) [inline spyder/plugins/base.py:514]:
```suggestion
                _("Lock pane to the current position"))
```

**Review by isabela-pf — COMMENTED** [review]
Hey there! I've seen this work a few times, and it looks like what I expected. Thank you so much for listing everything changed as a list.

I think this is good to merge from my end, I'd just like to suggest a minor change to the tool tips for specificity.

**Comment by ccordoba12** [comment]
@dalthviz, about your comments above:

> The new Unlock position is disabled for Plots

Fixed in one of my latest commits

> Seems like the Options menu now positions different if there is no space to show all the actions

That's most probably due to the addition of the new Lock/Unlock action because I didn't change that specifically. Other than this (which we could try to address in another PR), this should be ready.

**Pull request merged**